cosmic/tests/test_sample.py

```diff
@@ -88,34 +159,51 @@ def test_binary_fraction(self):
-        ecc = SAMPLECLASS.sample_ecc(ecc_model='uniform', size=10)
-        self.assertEqual(ecc.sum(), UNIFORM_ECC_SUM)
+        ecc = SAMPLECLASS.sample_ecc(ecc_model='uniform', size=100000)
+        slope = linear_fit(ecc)
+        self.assertEqual(np.round(slope, 1), FLAT_SLOPE)
 
-        np.random.seed(2)
+        np.random.seed(4)
         # Check that the sample_ecc function samples ecc properly
-        ecc = SAMPLECLASS.sample_ecc(ecc_model='sana12', size=10)
-        self.assertEqual(ecc.sum(), SANA12_ECC_SUM)
+        ecc = SAMPLECLASS.sample_ecc(ecc_model='sana12', size=1000000)
+        power_slope = power_law_fit(ecc)
+        self.assertEqual(np.round(power_slope, 2), SANA12_ECC_POWER_LAW)
 
-    def test_sample_porb_abt(self):
-        np.random.seed(2)
-        # Check that the sample_porb function samples porb properly
-        porb = SAMPLECLASS.sample_porb(1.0, 1.0, 0.5, 'log_uniform', size=1)
-        self.assertAlmostEqual(porb[0], PORB)
+    def test_sample_porb(self):
+        # next do Sana12
+        np.random.seed(4)
+        mass1, total_mass = SAMPLECLASS.sample_primary(primary_model='kroupa01', size=1000000)
+        mass2 = SAMPLECLASS.sample_secondary(primary_mass = mass1)
+        ecc = SAMPLECLASS.sample_ecc(ecc_model='sana12', size=len(mass1))
+        porb = SAMPLECLASS.sample_porb(mass1, mass2, ecc, 'sana12', size=len(mass1))
+        power_slope = power_law_fit(np.log10(porb))
+        self.assertAlmostEqual(np.round(power_slope, 2), SANA12_PORB_POWER_LAW)
 
-    def test_sample_porb_sana12(self):
-        np.random.seed(2)
-        porb = SAMPLECLASS.sample_porb(1.0, 1.0, 0.5, 'sana12', size=1)
-        self.assertAlmostEqual(porb[0], PORB_SANA12)
+        # next do Renzo19
+        np.random.seed(4)
+        porb_power = SAMPLECLASS.sample_porb(mass1+15, mass2, ecc, 'renzo19', size=len(mass1))
+        power_slope = power_law_fit(np.log10(porb_power))
+        self.assertEqual(np.round(power_slope, 2), SANA12_PORB_POWER_LAW)
+        porb_log_uniform = SAMPLECLASS.sample_porb(mass1, mass2, ecc, 'renzo19', size=len(mass1))
+        ind_log_uniform, = np.where(mass1 <= 15)
+        porb_log_uniform = porb_log_uniform[ind_log_uniform]
+        uniform_slope = linear_fit(np.log10(porb_log_uniform))
+        self.assertEqual(np.round(uniform_slope, 1), FLAT_SLOPE)
 
-    def test_sample_porb_renzo19(self):
         np.random.seed(2)
-        porb = SAMPLECLASS.sample_porb(1.0, 1.0, 0.5, 'renzo19', size=1)
-        self.assertAlmostEqual(porb[0], PORB_RENZO19)
+        # Check that the sample_porb function samples porb properly
+        porb = SAMPLECLASS.sample_porb(mass1, mass2, ecc, 'log_uniform', size=len(mass1))
+        # filter out the weird cuts for RLO and convert to sep
+        sep = a_from_p(porb, mass1, mass2)
+        sep = sep[sep > 10]
+        uniform_slope = linear_fit(np.log10(sep))
+        self.assertEqual(np.round(uniform_slope, 1), FLAT_SLOPE)
```

Comment:
Haven't gone through all these tests in much detail, but I trust that they're giving the desired results